### ISSUE TITLE
Mark classes / code as not threadsafe

### ISSUE THREAD

**Issue by hgschmie** [spotbugs/spotbugs#3390]
Hi,

We use spotbugs in a largeish project and there are a number of false positives where spotbugs reports problems with thread (AT_NONATOMIC_64BIT_PRIMITIVE, AT_STALE_THREAD_WRITE_OF_PRIMITIVE, AT_NONATOMIC_OPERATIONS_ON_SHARED_VARIABLE). 

The problem is that the places where this is reported have been explicitly documented as not threadsafe. So problems may exist but it does not matter because using this code from multiple threads is a problem in general.

Is there a way to mark such classes explicitly as not threadsafe so that spotbugs does not try to find multithreading bugs in code that is not intended to be used from multiple threads?

**Comment by JuditKnoll** [comment]
You can suppress these bugs with the `@SuppressFBWarnings` annotation in the code or you can use a [filter file](https://spotbugs.readthedocs.io/en/latest/filter.html#) to exclude them.
If you can share a minimal example, the detector can be modified in SpotBugs to not report these false positives, if necessary.

**Comment by stevenschlansker** [comment]
Thanks @JuditKnoll we are aware we can suppress the warnings, however they are so numerous that we would probably just turn these checks off entirely.

Here is an example false positive:

```
import javax.annotation.concurrent.NotThreadSafe;

@NotThreadSafe
public class CountingRunner implements Runnable {
    private int counter;

    public int getCount() {
        return counter;
    }

    @Override
    public void run() {
        counter++;
    }
}
```

fails:
```
[ERROR] Medium: Operation on the "counter" shared variable in "CountingRunner" class is not atomic [CountingRunner] At CountingRunner.java:[line 15] AT_NONATOMIC_OPERATIONS_ON_SHARED_VARIABLE
```

**Comment by stevenschlansker** [comment]
Perhaps updating `MultiThreadedCodeIdentifierUtils.isPartOfMultiThreadedCode` to return `false` for classes annotated as not thread safe would fix this?
